pypxe/dhcp.py

```diff
@@ -37,30 +37,43 @@ def __init__(self, **serverSettings):
         self.ipxe = serverSettings.get('useipxe', False)
         self.http = serverSettings.get('usehttp', False)
         self.mode_proxy = serverSettings.get('mode_proxy', False) #ProxyDHCP mode
-        self.mode_debug = serverSettings.get('mode_debug', False) #debug mode
+        self.logger =  serverSettings.get('logger', None)
+        self.mode_debug =  serverSettings.get('mode_debug', False)
         self.magic = struct.pack('!I', 0x63825363) #magic cookie
 
+        if self.logger == None:
+            import logging
+            import logging.handlers
+            # setup logger
+            self.logger = logging.getLogger("dhcp")
+            handler = logging.StreamHandler()
+            formatter = logging.Formatter('%(asctime)s %(levelname)s %(message)s')
+            handler.setFormatter(formatter)
+            self.logger.addHandler(handler)
+
+            if self.mode_debug:
+                self.logger.setLevel(logging.DEBUG)
+
         if self.http and not self.ipxe:
-            print '\nWARNING: HTTP selected but iPXE disabled. PXE ROM must support HTTP requests.\n'
+            self.logger.warning('HTTP selected but iPXE disabled. PXE ROM must support HTTP requests.')
         if self.ipxe and self.http:
             self.filename = 'http://{fileserver}/{filename}'.format(fileserver = self.fileserver, filename = self.filename)
         if self.ipxe and not self.http:
             self.filename = 'tftp://{fileserver}/{filename}'.format(fileserver = self.fileserver, filename = self.filename)
 
-        if self.mode_debug:
-            print 'NOTICE: DHCP server started in debug mode. DHCP server is using the following:'
-            print '\tDHCP Server IP: {}'.format(self.ip)
-            print '\tDHCP Server Port: {}'.format(self.port)
-            print '\tDHCP Lease Range: {} - {}'.format(self.offerfrom, self.offerto)
-            print '\tDHCP Subnet Mask: {}'.format(self.subnetmask)
-            print '\tDHCP Router: {}'.format(self.router)
-            print '\tDHCP DNS Server: {}'.format(self.dnsserver)
-            print '\tDHCP Broadcast Address: {}'.format(self.broadcast)
-            print '\tDHCP File Server IP: {}'.format(self.fileserver)
-            print '\tDHCP File Name: {}'.format(self.filename)
-            print '\tProxyDHCP Mode: {}'.format(self.mode_proxy)
-            print '\tUsing iPXE: {}'.format(self.ipxe)
-            print '\tUsing HTTP Server: {}'.format(self.http)
+        self.logger.debug('NOTICE: DHCP server started in debug mode. DHCP server is using the following:')
```

Comment:
In L64-L76 should we remove the `\t` characters since we're using the logger now?